spec/features/trainee_actions/change_course_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Change course", type: :feature, feature_publish_course_details: true do
+  include SummaryHelper
+
+  before do
+    given_i_am_authenticated
```

Comment:
The naming of these steps looks a bit akward when you put it all together:

```
given_i_am_authenticated
given_a_trainee_exists_with_trn_received
given_trainee_related_courses_exist
and_i_am_on_the_trainee_record_page
given_i_click_to_change_course_details
and_i_choose_a_different_course
and_i_click_continue
and_select_a_specialism_if_necessary
when_i_click_update_record
then_the_trainee_course_has_changed
```

Feels like it should be re-worded to something:

```
given_i_am_authenticated
and_a_trainee_exists_with_trn_received
and_a_trainee_with_related_courses_exist
and_i_am_on_the_trainee_record_page
when_i_click_to_change_course_details
and_i_choose_a_different_course
and_i_click_continue
and_select_a_specialism_if_necessary
and_i_click_update_record
then_the_trainee_course_has_changed
```

Reply:
Fair point. I was anticipating other scenarios getting added in the future, so I decided to push some common steps into the background task - we do this is other places. I'm happy to get rid of the background task and make the story flow better.

I'm not a fan of using methods beginnng with `given`, `when`, `then` because it becomes harder to construct a well meaning story. I think cucumber does a better job using regexes and `Given`, `When`, `Then` blocks. Anyway, that boat has sailed long ago. :)